uvicorn/protocols/http/h11_impl.py

```diff
@@ -266,6 +266,11 @@ def handle_upgrade(self, event):
         if upgrade_value != b"websocket" or self.ws_protocol_class is None:
             msg = "Unsupported upgrade request."
             self.logger.warning(msg)
+            from uvicorn.protocols.websockets.auto import AutoWebSocketsProtocol
+
+            if AutoWebSocketsProtocol is None:
+                msg = "You have no ws libary installed, either use pip install uvicorn[standard] or install websockets / wsproto depending on the ws library you want to use."  # noqa: E501
```

Comment:
Small suggestion on the copywriting:

```suggestion
                msg = "No supported WebSocket library detected. Please use 'pip install uvicorn[standard]', or install 'websockets' or 'wsproto' manually."  # noqa: E501
```